AgendaUWP/AgendaUWP/ViewModels/ContactListPageViewModel.cs

```diff
@@ -1,27 +1,89 @@
-﻿using Prism.Windows.Mvvm;
+﻿using AgendaUWP.Models;
+using Prism.Commands;
+using Prism.Windows.Mvvm;
+using Prism.Windows.Navigation;
 using Service.Contact;
 using Service.Model;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Linq;
 
 namespace AgendaUWP.ViewModels
 {
     public class ContactListPageViewModel : ViewModelBase
     {
+
+        #region properties
+
         private readonly IContactService<Contact> _contactService;
+        private INavigationService _navigationService;
+
+        private ObservableCollection<ContactItemsGroup> data;
+        public ObservableCollection<ContactItemsGroup> Data
+        {
+            get { return data; }
+            set { SetProperty(ref data, value); }
+        }
+
+        public IEnumerable<ContactItemsGroup> GroupedContactItems { get; private set; }
 
-        public ContactListPageViewModel(IContactService<Contact> contactService)
+        #endregion
+
+        #region constructors
+
+        public ContactListPageViewModel(IContactService<Contact> contactService, INavigationService navigationService)
         {
             this._contactService = contactService;
-            GetData();
+            this._navigationService = navigationService;
+            Data = GetData();
+            InitilizeCommands();
         }
 
-        private void GetData()
+        #endregion
+
+        #region commands
+
+        public DelegateCommand<Contact> DetailCommand { get; private set; }
+
+        #endregion
+
+        #region initializers
+
+        private void InitilizeCommands()
         {
-            var contacts = _contactService.GetContacts();
-            GroupedContactItems = contacts.GroupBy(m => m.FullName[0] , (key, list) => new ContactItemsGroup(key, list));
+            DetailCommand = new DelegateCommand<Contact>(GoToDetailContact);
         }
 
-        public IEnumerable<ContactItemsGroup> GroupedContactItems { get; private set; }
+        #endregion
+
+        #region methods
+
+        private void GoToDetailContact(Contact contact)
+        {
+            _navigationService.Navigate(PageTokens.DetailContactPage, contact);
+        }
+
+        private ObservableCollection<ContactItemsGroup> GetData()
+        {
+            var contacts = _contactService.GetContacts();
+
+            ObservableCollection<ContactItemsGroup> contactsObservable;
+
+            if (contacts != null && contacts.Count() > 0)
```

Comment:
Please consider to remove it and returns an empty list in ContactService. To avoid check if it's null.

Reply:
done.